### ISSUE TITLE
xgc-restart HDF5 tests fail to read

### ISSUE THREAD

**Issue by chuckatkins** [ornladios/ADIOS2-Testing#32]
The adios2-iotest based tests are unable to execute the HDF5 reader because of an improper filename being used.  While the test configuration explicitly uses `xgc.restart.00040.bp`, the HDF5 writer automatically replaces `.bp` with `.h5`but the reader does no such substitution so it tries to open the `.bp` file which doesn't exist.

**Comment by chuckatkins** [comment]
I would support removing the auto extension change in the writer since there's plenty of files that use the HDF5 format that don't use it (exodus, cgns, etc.).  ADIOS being a lower level library should allow a user to use whatever filename they want.  I'll open a corresponding https://github.com/ornladios/adios2 issue with this.

**Comment by guj** [comment]
Yes, it was an old request to be "smart" about file extensions.  The IO class also tries to infer which type of Engine to use by looking at filename extension.   That needs to be changed  as well then. @pnorbert what do you think?
